functions/_fzf_search_completions.fish

```diff
@@ -0,0 +1,189 @@
+# External limitations:
+# - fish-shell/fish-shell#9577: folders ending in '=' break fish's internal path completion
+#   (It thinks that it has to complete from / because `a_folder=/` looks like an argument to fish.
+# - fish cannot give reliable context information on completions. Knowing whether a completion
+#   is a file or argument can only be determined via a heuristic.
+# - Completion is slow. Not because this script is slow but because fish's `complete` command
+#   can take several seconds even for just 100 entries. Just run `time complete --escape --do-complete l`
+#   to see for yourself.
+set --global _fzf_search_completions_min_description_offset 14
+function _fzf_search_completions --description "Shell completion using fzf"
+	# Produce a list of completions from which to choose
+	# and do nothing if there is nothing to select from
+	set -l cmd (commandline --cut-at-cursor --tokenize) (commandline --current-token)
+	set -l completions (complete --escape --do-complete "$cmd")
+	test (count $completions) -eq 0; and return
+
+	set -l results
+	set -l first_completion (string split --fields 1 --max 1 --right \t -- $completions[1])
+
+	# A prefix that will be prepended before each selected completion
+	set -l each_prefix ""
+
+	# Only skip fzf if there is a single completion and it starts with the expected completion prefix.
```

Comment:
WDYM "starts with the expected completion prefix"? why would `complete --do-complete` ever return a completion the user wouldn't want?

Reply:
Sometimes the shell transforms or eliminates quotes, variables, ...
If that happens, I don't want this to silently change the commandline and instead prompt the user with fzf so any change is previewed and can be cancelled if needed.

Consider `ls foo"bar"` could be completed as `foobar`, but the user might have intended to place quotes there to e.g. later insert content with spaces more easily. I often do that and it's annoying if <TAB> deletes the quotes.

Reply:
Ah okay this is consistent with how fish currently operates: if only one completion, it will auto-complete while preserving existing quotes and backslashes. Makes sense, good thinking!